NDTensors/src/lib/GradedAxes/src/fusion.jl

```diff
@@ -18,26 +18,40 @@ function tensor_product(
   return foldl(tensor_product, (a1, a2, a3, a_rest...))
 end
 
-function tensor_product(a1::AbstractUnitRange, a2::AbstractUnitRange)
+function tensor_product(::AbstractUnitRange, ::AbstractUnitRange)
   return error("Not implemented yet.")
 end
 
 function tensor_product(a1::Base.OneTo, a2::Base.OneTo)
   return Base.OneTo(length(a1) * length(a2))
 end
 
-function tensor_product(a1::OneToOne, a2::AbstractUnitRange)
+function tensor_product(::OneToOne, a2::AbstractUnitRange)
   return a2
 end
 
-function tensor_product(a1::AbstractUnitRange, a2::OneToOne)
+function tensor_product(a1::AbstractUnitRange, ::OneToOne)
   return a1
 end
 
-function tensor_product(a1::OneToOne, a2::OneToOne)
+function tensor_product(::OneToOne, ::OneToOne)
   return OneToOne()
 end
 
+# Handle dual. Always return a non-dual GradedUnitRange.
+function tensor_product(a1::AbstractUnitRange, a2::UnitRangeDual)
+  return tensor_product(a1, flip(a2))
+end
+
+function tensor_product(a1::UnitRangeDual, a2::AbstractUnitRange)
+  return tensor_product(flip(a1), a2)
+end
+
+# TBD change convention to tensor(dual, dual) -> dual?
+function tensor_product(a1::UnitRangeDual, a2::UnitRangeDual)
+  return tensor_product(flip(a1), flip(a2))
+end
```

Comment:
I realized that these can all be combined into a single definition by defining a `flip_dual` function:
```julia
flip_dual(r::AbstractUnitRange) = r
flip_dual(r::UnitRangeDual) = flip(r)
function tensor_product(a1::AbstractUnitRange, a2::AbstractUnitRange)
  return tensor_product(flip_dual(a1), flip_dual(a2))
end
```
Maybe there is a better name than `flip_dual` but you get the idea, it only calls `flip` if the space is a dual space.

Once more dual unit range types are introduced in #1531, probably we will want to turn these into trait functions as discussed in https://github.com/ITensor/ITensors.jl/pull/1363/files#r1779660676, but we can leave that for #1531.